File: JAVA/Day5/TestMiniParkingSystem.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class MiniParkingSystemTest {

    @Test
    void testParkCar() {
        MiniParkingSystem.resetParking();
        String result = MiniParkingSystem.parkCar();
        assertEquals("Your car has been parked at 1", result);
        assertEquals(1, MiniParkingSystem.getParking()[0]);
    }

    @Test
    void testRemoveCar() {
        MiniParkingSystem.resetParking();
        MiniParkingSystem.parkCar();
        String result = MiniParkingSystem.removeCar(1);
        assertEquals("Car removed from slot 1", result);
        assertEquals(0, MiniParkingSystem.getParking()[0]);
    }

    @Test
    void testRemoveEmptySlot() {
        MiniParkingSystem.resetParking();
        String result = MiniParkingSystem.removeCar(1);
        assertEquals("The parking space at 1 is already empty, so there is space for you to park into.", result);
    }

    @Test
    void testInvalidSlot() {
        MiniParkingSystem.resetParking();
        String result = MiniParkingSystem.removeCar(21);
        assertEquals("Invalid car parking space number!", result);
    }

    @Test
    void testParkingFull() {
        MiniParkingSystem.resetParking();
        for (int i = 0; i < 20; i++) {
            MiniParkingSystem.parkCar();
        }
        String result = MiniParkingSystem.parkCar();
        assertEquals("Sorry! The Parking lot is currently full, you can opt to drive out some to repark another.", result);
    }
}

File: Python/Day5/Test_Mini_Parking_System.py
import unittest
import Mini_Parking_System

class Test_Mini_Parking(unittest.TestCase):

    def setUp(self):
        Mini_Parking_System.reset_parking()

    def test_park_car(self):
        result = Mini_Parking_System.park_car()
        self.assertEqual("Your car has been parked at 1", result)
        self.assertEqual(1, Mini_Parking_System.getParking()[0])

    def test_parking_full(self):
        for _ in range(20):
            Mini_Parking_System.park_car()
        result = Mini_Parking_System.park_car()
        self.assertEqual("Sorry! Parking lot is full.", result)

    def test_remove_car(self):
        Mini_Parking_System.park_car()
        result = Mini_Parking_System.remove_car(1)
        self.assertEqual("Car removed from slot 1", result)
        self.assertEqual(0, Mini_Parking_System.getParking()[0])

    def test_invalid_remove(self):
        result = Mini_Parking_System.remove_car(25)
        self.assertEqual("Invalid slot number!", result)

